clippy_lints/src/ranges.rs

```diff
@@ -53,11 +54,12 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for StepByZero {
 
             // Range with step_by(0).
             if name == "step_by" && args.len() == 2 && has_step_by(cx, &args[0]) && is_integer_literal(&args[1], 0) {
```

Comment:
I wonder why we are not evaluating the constant here rather than linting on literal `0`.